src/utils/ParseRouteUtils.js

```diff
@@ -142,85 +136,79 @@ async function condenseRoute(
   departureDelayBuffer: ?boolean = false,
   departureTimeNowMs: number,
 ) {
-  try {
-    // trim unnecessary walking directions
-    await trimFirstLastDirections(route, startCoords, endCoords);
+  // trim unnecessary walking directions
+  await trimFirstLastDirections(route, startCoords, endCoords);
+
+  let previousDirection = null;
+  let totalDistanceWalking = 0;
+
+  for (let index = 0; index < route.directions.length; index++) {
```

Comment:
isn't forEach a thing for looping like this? you don't really use **index** for anything except the beginning

Reply:
it's used on line 187, I didn't want to break existing functionality but I agree that this should be changed in the future, I'll make an issue for it!